pkg/bitnet/internal/model/tokenizer.go

```diff
@@ -0,0 +1,162 @@
+package model
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
```

Comment:
The same thing in this implementation. We shouldn't hard code to use real os filesystems, but instead take filesystem as a dependency which would also support embedded filesystems.